### DISCUSSION TITLE
Scripting a thermostat with Timer

### DISCUSSION THREAD

**Discussion by javier6d2** [arendst/Tasmota#13366] [Q&A] [answered]
I am wondering if would be possible to toggle a thermostat (switch based on temp) when a global Timer is activated using scripting feature. Using rules would be like this:

```
Rule1
  ON tele-DS18B20#Temperature<25 DO Power1 1 ENDON
  ON tele-DS18B20#Temperature>30 DO Power1 0 ENDON

Rule1 1 
```

Rule2 trigger Rule1.  
Rule1 is triggered when Timer1 starts (in the morning) and stops when Timer2 starts (in the evenings). 

```
Rule2
  ON Clock#Timer=1 DO Rule1 1 ENDON
  ON Clock#Timer=2 DO Rule1 0 ENDON

Rule2 1
```

**Comment by barbudor** [comment, marked as answer]
When using Scripting (instead of rules) you will get an event `>E` triggered 
I don't know enough how to process

**Comment by javier6d2** [comment]
Thank you @barbudor, I dont know also how to process this. May be @gemu2015 can put same light.

**Comment by gemu2015** [comment]
small script to start with

\>D
tmp=0
\>T
tmp=DS18B20#Temperature

\>S
; time contains minutes from midnight
; from 8 to 20 hours
if time>480
and time<1200
then

if tmp<25
then
->power1 1
endif

if tmp>30
then
->power1 0
endif

endif

**Comment by javier6d2** [comment]
Thank you @gemu2015. I believe that with this help I can do it
